dataRetriever/blockchain/metachain.go

```diff
@@ -53,6 +54,7 @@ func (mc *metaChain) SetGenesisHeader(header data.HeaderHandler) error {
 }
 
 // SetCurrentBlockHeaderAndRootHash sets current block header pointer and the root hash
+// Question for review: perhaps also receive the blockHash as a parameter (since SetCurrentBlockHeaderAndRootHash and SetCurrentBlockHeaderHash are always called in pair, so that we set all of them in the same critical section)?
```

Comment:
:+1:

Reply:
Even though they were always called in pair in production code, they weren't in some tests. In order to avoid breaking the tests, I did not apply this change at the moment. Is this all right?